interpreter/exec/ixx.ml

```diff
@@ -160,8 +199,15 @@ struct
   let shr_s x y =
     shift Rep.shift_right x y
 
+  (* Check if we are storing smaller ints. *)
+  let needs_extend = shl one (Rep.of_int (Rep.bitwidth - 1)) <> Rep.min_int
+
   let shr_u x y =
-    shift Rep.shift_right_logical x y
+    (* If we are storing smaller ints, we need to mask out the high bits. *)
```

Comment:
We don't have that problem because all the SIMD operations pass through `to_lanes` and `of_lanes`, which strips the junk bits and appropriately extends to i32, so later on the shr_s works correctly.

I.e. i8.ml itself is broken, but it's usage together with v128.ml is fine.

`shr_u` requires this fix (i think) because it is used in `of_string`.

Reply:
Ah, I see, good. But I still feel uncomfortable with having broken implementations of the I8/I16 modules themselves. It is likely that somebody else is going to try using them in the future for purposes where this problem is not masked.

Reply:
Fixed in https://github.com/WebAssembly/simd/pull/528